pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -376,15 +380,15 @@ private void processKafkaEvents(Iterable<MessageAndOffset> messagesAndOffsets, L
         transformedRow = _fieldExtractor.transform(decodedRow, transformedRow);
 
         if (transformedRow != null) {
-          _serverMetrics.addMeteredTableValue(_metricKeyName, ServerMeter.REALTIME_ROWS_CONSUMED, 1);
+          realtimeRowsConsumedMeter = _serverMetrics.addMeteredTableValue(_metricKeyName, ServerMeter.REALTIME_ROWS_CONSUMED, 1, realtimeRowsConsumedMeter);
```

Comment:
We could do it for every batch of rows that we get from Kafka. I looked at the underlying code and it just does a CAS on a long value (actually it does 4, as it keeps track of the 1/5/15 minute rates as well), so it's not that expensive.

I still need to review our usage of other metrics and update them if necessary to use this newer API, as well as probably change the API a bit for the metrics, so I'll keep that in mind when I revisit that part.